pkg/challenges/challenges.go

```diff
@@ -205,6 +206,69 @@ func GithubWorkflow(ctx context.Context, principal *oidc.IDToken, pubKey crypto.
 	}, nil
 }
 
+func URI(ctx context.Context, principal *oidc.IDToken, pubKey crypto.PublicKey, challenge []byte) (*ChallengeResult, error) {
+	uriWithSubject := principal.Subject
+
+	cfg, ok := config.FromContext(ctx).GetIssuer(principal.Issuer)
+	if !ok {
+		return nil, errors.New("invalid configuration for OIDC ID Token issuer")
+	}
+
+	uSubject, err := url.Parse(uriWithSubject)
+	if err != nil {
+		return nil, err
+	}
+
+	// The subject prefix URI must match the domain (excluding the subdomain) of the issuer
+	// In order to declare this configuration, a test must have been done to prove ownership
+	// over both the issuer and domain configuration values.
+	// Valid examples:
+	// * uriWithSubject = https://domain.com/users/user1, issuer = https://accounts.domain.com
+	// * uriWithSubject = https://accounts.domain.com/users/user1, issuer = https://accounts.domain.com
+	// * uriWithSubject = https://users.domain.com/users/user1, issuer = https://accounts.domain.com
+	uIssuer, err := url.Parse(cfg.IssuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that:
+	// * The URI schemes match
+	// * Either the hostnames exactly match or the top level and second level domains match
+	if err := isURISubjectAllowed(uSubject, uIssuer); err != nil {
+		return nil, err
+	}
+
+	// The subject hostname must exactly match the subject domain from the configuration
+	uDomain, err := url.Parse(cfg.SubjectDomain)
+	if err != nil {
+		return nil, err
+	}
+	if uSubject.Scheme != uDomain.Scheme {
+		return nil, fmt.Errorf("Subject URI scheme (%s) must match expected domain URI scheme (%s)", uSubject.Scheme, uDomain.Scheme)
```

Comment:
Nit: I thought error messages starting with capital letters weren't considered idiomatic? This shows again in several other `fmt.Errorf`s in the file.

Reply:
My C++ development is showing :) Fixed